sound/soc/sof/Makefile

```diff
@@ -24,3 +24,4 @@ obj-$(CONFIG_SND_SOC_SOF_INTEL_TOPLEVEL) += intel/
 obj-$(CONFIG_SND_SOC_SOF_IMX_TOPLEVEL) += imx/
 obj-$(CONFIG_SND_SOC_SOF_AMD_TOPLEVEL) += amd/
 obj-$(CONFIG_SND_SOC_SOF_XTENSA) += xtensa/
+obj-$(CONFIG_SND_SOC_SOF_MTK_TOPLEVEL) += snd-sof.o  mediatek/
```

Comment:
Why you need to add here `snd-sof.o`? Wouldn't `obj-$(CONFIG_SND_SOC_SOF_MTK_TOPLEVEL) +=  mediatek/` be enough?

Reply:
I set CONFIG_SND_SOC_SOF_OF=m which produce snd-sof-of.ko so that we can use install module for it. Therefore I should add snd-sof.o to resolve symbol undefined errors.

Reply:
@yaochunhung Make your own Mediatek thing depend on it in the Kconfig so that it will also be built as a module. That way you shouldn't have an issue.

Reply:
yes, that's not good and probably the reason why the build it broken.

Reply:
things work fine for me without this 

````diff
diff --git a/sound/soc/sof/Makefile b/sound/soc/sof/Makefile
index 7cadd653a9bb..43a983376976 100644
--- a/sound/soc/sof/Makefile
+++ b/sound/soc/sof/Makefile
@@ -24,4 +24,4 @@ obj-$(CONFIG_SND_SOC_SOF_INTEL_TOPLEVEL) += intel/
 obj-$(CONFIG_SND_SOC_SOF_IMX_TOPLEVEL) += imx/
 obj-$(CONFIG_SND_SOC_SOF_AMD_TOPLEVEL) += amd/
 obj-$(CONFIG_SND_SOC_SOF_XTENSA) += xtensa/
-obj-$(CONFIG_SND_SOC_SOF_MTK_TOPLEVEL) += snd-sof.o  mediatek/
+obj-$(CONFIG_SND_SOC_SOF_MTK_TOPLEVEL) += mediatek/
````

Reply:
@plbossart I also build fine in my local test with this PR. I will modify it. Thanks.